builtin/providers/triton/resource_firewall_rule.go

```diff
@@ -0,0 +1,109 @@
+package triton
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/joyent/gosdc/cloudapi"
+)
+
+func resourceFirewallRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceFirewallRuleCreate,
+		Exists: resourceFirewallRuleExists,
+		Read:   resourceFirewallRuleRead,
+		Update: resourceFirewallRuleUpdate,
+		Delete: resourceFirewallRuleDelete,
+
+		Schema: map[string]*schema.Schema{
+			// required
+			"rule": {
+				Description: "firewall rule text",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+
+			// optional
+			"enabled": {
+				Description: "Indicates if the rule is enabled",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
```

Comment:
IMO this should mirror the Joyent default - this is what we tend to do with other providers AFAICT